From 3e36c99d401530dc026712523a7c445da17ee299 Mon Sep 17 00:00:00 2001
From: Marek Blaha <mblaha@redhat.com>
Date: Mon, 14 Mar 2022 09:49:52 +0100
Subject: [PATCH] Fix processing of download errors (RhBug: 2024527)

Users with different than english locale are not able to update their
systems in case that some of updates are already downloaded in the dnf
cache (e.g. using dnf-automatic).

The error string is taken from librepo target where it is stored
untranslated. Therefore we need to compare untranslated versions of the
string.

= changelog =
msg:           Fix download errors handling in non-english locales
type:          bugfix
resolves:      https://bugzilla.redhat.com/show_bug.cgi?id=2024527
---
 dnf/repo.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dnf/repo.py b/dnf/repo.py
index 1822cf0..ec1a253 100644
--- a/dnf/repo.py
+++ b/dnf/repo.py
@@ -108,7 +108,7 @@ def _download_payloads(payloads, drpm, fail_fast=True):
         callbacks = tgt.getCallbacks()
         payload = callbacks.package_pload
         pkg = payload.pkg
-        if err == _('Already downloaded'):
+        if err == 'Already downloaded':
             errs._skipped.add(pkg)
             continue
         pkg.repo._repo.expire()
--
libgit2 1.1.0

